File: is_looping/is_looping.py
class Node:
    def __init__(self, val):
        self.val = val
        self.next = None

node1 = Node(12)
node2 = Node(99)
node3 = Node(30)
node4 = Node(30)
node5 = Node(30)
node6 = Node(30)
node7 = Node(30)
node8 = Node(30)
node9 = Node(30)

node1.next = node2
node2.next = node3
node3.next = node4
node4.next = node5
node5.next = node6
node6.next = node7
node7.next = node8
node8.next = node9
node9.next = node3

def is_looping(head):
    slow = head
    fast = head
    while (slow and fast and fast.next):
        slow = slow.next
        fast = fast.next.next
        if (slow == fast):
            return 1
    return 0

print(is_looping(node1))
    


    

File: 2/is_looping/is_looping.c
struct s_node {
	int value;
	struct s_node *next;
};

int	is_looping(struct s_node *node)
{
	struct s_node *fast = node;
	struct s_node *slow = node;

	while (node)
	{
		slow = slow->next;
		if (!fast->next)
			return (0);
		if (!fast->next->next)
			return (0);
		fast = fast->next->next;
		if (slow == fast)
			return (1);
		node = node->next;
	}
	return (0);
}
/*
#include <stdlib.h>
#include <stdio.h>
int main()
{
	struct s_node *a = malloc(sizeof(struct s_node));
	struct s_node *b = malloc(sizeof(struct s_node));
	struct s_node *c = malloc(sizeof(struct s_node));
	struct s_node *d = malloc(sizeof(struct s_node));
	struct s_node *e = malloc(sizeof(struct s_node));
	struct s_node *f = malloc(sizeof(struct s_node));
	a->value = 1;
	b->value = 2;
	c->value = 3;
	d->value = 4;
	e->value = 5;
	f->value = 6;
	a->next = b;
	b->next = c;
	c->next = d;
	d->next = e;
	e->next = f;
	f->next = e;
	printf("%i\n", is_looping(a));
}*/
